src/app.rs

```diff
@@ -158,13 +158,11 @@ impl<'a, 'b> Application<'a, 'b> {
             #[cfg(feature = "profiler")]
             profile_scope!("handle_event");
 
-            let events = match world
+            let events = world
                 .read_resource::<EventChannel<Event>>()
-                .lossy_read(&mut self.events_reader_id)
-            {
-                Ok(data) => data.cloned().collect(),
-                _ => Vec::default(),
-            };
+                .read(&mut self.events_reader_id)
+                .cloned()
+                .collect::<Vec<_>>();
 
             for event in events {
```

Comment:
Maybe use the `StorageIterator` that the event channel gets in the loop below?

Reply:
Can't because of borrow rules on world

Reply:
Ah true